controllers/storagecluster/storageclient.go

```diff
@@ -44,7 +44,11 @@ func (s *storageClient) ensureCreated(r *StorageClusterReconciler, storagecluste
 	storageClient.Name = storagecluster.Name
 	_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, storageClient, func() error {
 		if storageClient.Status.ConsumerID == "" {
-			token, err := util.GenerateClientOnboardingToken(tokenLifetimeInHours, onboardingPrivateKeyFilePath, nil)
+			privateKey, err := util.GetParsedPrivateKey(r.Client, r.OperatorNamespace)
```

Comment:
instead of OperatorNamespace we should be using storageCluster Namespace as private key pair is generated per storageCluster and it is created in the same namespace as the storageCluster

Reply:
I don't think so, could you pls expand more?

Reply:
Onboarding job is created in the same [namespace](https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/provider_server.go#L385) as the storageCluster. The job generates the private/pub key pair in the same [namespace](https://github.com/red-hat-storage/ocs-operator/blob/main/onboarding-validation-keys-generator/main.go#L34) as the storageCluster is created. If the storageCluster is created in a different namespace from operator namespace, it will fail at this condition. So it is better if we use storageCluster namespace instead of Operator namespace

Reply:
ack, changed to storagecluster namespace